src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/ClusterDetailFragment.kt

```diff
@@ -19,35 +44,181 @@ import kotlinx.coroutines.cancel
  * ClusterDetailFragment allows user to pick cluster, command, specify parameters and see
  * the callback result.
  */
-class ClusterDetailFragment : Fragment(){
+class ClusterDetailFragment : Fragment() {
   private val deviceController: ChipDeviceController
     get() = ChipClient.getDeviceController(requireContext())
 
   private val scope = CoroutineScope(Dispatchers.Main + Job())
+  private lateinit var clusterMap: HashMap<String, ClusterInfo>
+  private lateinit var selectedClusterInfo: ClusterInfo
+  private lateinit var selectedCluster: ChipClusters.BaseChipCluster
+  private lateinit var selectedCommandCallback: DelegatedClusterCallback
+  private var commandArguments = HashMap<String, Any>()
+  private lateinit var selectedCommandInfo: CommandInfo
+  private var devicePtr = 0L
+  private var endpointId = 1
```

Comment:
This should be passed based on the endpoint selected in `ClusterInteractionFragment`